test/extensions/quic_listeners/quiche/integration/quic_http_integration_test.cc

```diff
@@ -218,5 +228,138 @@ TEST_P(QuicHttpIntegrationTest, TestDelayedConnectionTeardownTimeoutTrigger) {
+  for (size_t i = 1; i <= concurrency_; ++i) {
+    // The BPF filter looks at the 1st byte of connection id in the packet
+    // header. And currently all QUIC versions support 8 bytes connection id. So
+    // create connections with the first 4 bytes of connection id different from each
+    // other so they should be evenly distributed.
+    cached->add_server_designated_connection_id(quic::test::TestConnectionId(i << 32));
+    codec_clients.push_back(makeHttpConnection(lookupPort("http")));
+  }
+  if (GetParam() == Network::Address::IpVersion::v4) {
+    test_server_->waitForCounterEq("listener.0.0.0.0_0.downstream_cx_total", 8u);
+  } else {
+    test_server_->waitForCounterEq("listener.[__]_0.downstream_cx_total", 8u);
+  }
+  // Even without BPF support, these connections should more or less distributed
+  // across different workers.
+  for (size_t i = 0; i < concurrency_; ++i) {
+    if (GetParam() == Network::Address::IpVersion::v4) {
+      EXPECT_LT(
+          test_server_->gauge(fmt::format("listener.0.0.0.0_0.worker_{}.downstream_cx_active", i))
+              ->value(),
+          8u);
+      EXPECT_LT(
+          test_server_->counter(fmt::format("listener.0.0.0.0_0.worker_{}.downstream_cx_total", i))
+              ->value(),
+          8u);
+    } else {
+      EXPECT_LT(
+          test_server_->gauge(fmt::format("listener.[__]_0.worker_{}.downstream_cx_active", i))
+              ->value(),
+          8u);
+      EXPECT_LT(
+          test_server_->counter(fmt::format("listener.[__]_0.worker_{}.downstream_cx_total", i))
+              ->value(),
+          8u);
+    }
+  }
+  for (size_t i = 0; i < concurrency_; ++i) {
+    codec_clients[i]->close();
+  }
+#ifdef SO_ATTACH_REUSEPORT_CBPF_TMP
+#define SO_ATTACH_REUSEPORT_CBPF SO_ATTACH_REUSEPORT_CBPF_TMP
+#endif
+}
+
+TEST_P(QuicHttpIntegrationTest, ConnectionMigration) {
+#if defined(SO_ATTACH_REUSEPORT_CBPF) && defined(__linux__)
+  concurrency_ = 2;
+  set_reuse_port_ = true;
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto encoder_decoder =
+      codec_client_->startRequest(Http::TestHeaderMapImpl{{":method", "POST"},
+                                                          {":path", "/test/long/url"},
+                                                          {":scheme", "http"},
+                                                          {":authority", "host"}});
+  request_encoder_ = &encoder_decoder.first;
+  auto response = std::move(encoder_decoder.second);
+
+  codec_client_->sendData(*request_encoder_, 1024u, false);
+
+  // Change to a new port by switching socket, and connection should still continue.
+  Network::Address::InstanceConstSharedPtr local_addr =
+      Network::Test::getCanonicalLoopbackAddress(version_);
+  quic_connection_->switchConnectionSocket(
+      createConnectionSocket(server_addr_, local_addr, nullptr));
```

Comment:
Are there any stats we can check to make sure this actually migrated, and that we didn't somehow end up with a testing no-op?